apps/client/src/app/pages/voyage-tracker/voyage-tracker/vessel-row/vessel-build-column/vessel-build-column.component.ts

```diff
@@ -58,4 +58,12 @@ export class VesselBuildColumnComponent {
   get condition(): string {
     return Object.keys(this.parts).map((slot) => `${((this.parts[slot].condition || 0) / 300).toFixed(2)}%`).join(' - ');
   }
+
+  conditionClass(): object {
+    const worstCondition = Math.min(...Object.keys(this.parts).map((slot) => (this.parts[slot].condition || 0) / 300));
+    return {'good': worstCondition > 50,
+            'caution': worstCondition <= 50 && worstCondition > 25,
+            'warn': worstCondition <= 25 && worstCondition > 0,
+            'broken': worstCondition <= 0}
+  }
```

Comment:
Instead of this, you'll want to use an `Observable`, because observables are only triggered when the value changes, making your code efficient for this, because you don't need to update condition class if parts didn't change at all.

Something like:

```ts
conditionClass$ = this.parts$.pipe(
                                              map(parts => {
                                                  const worstCondition = Math.min(...Object.keys(this.parts).map((slot) => (this.parts[slot].condition || 0) / 300));
                                                  return {'good': worstCondition > 50,
                                                              'caution': worstCondition <= 50 && worstCondition > 25,
                                                              'warn': worstCondition <= 25 && worstCondition > 0,
                                                              'broken': worstCondition <= 0}
                                                  })
                               );
```

Right after the `parts$` observable and then use it in template by doing `conditionClass$ | async` instead of `conditionClass()`.